nodejs/server.js

```diff
@@ -0,0 +1,254 @@
+        response.status(200).json(res.json[0]);
+    })
+});
+
+// insert new post
+app.post('/posts', (req, response) => {
+    console.log(req.method + ' ' + req.url);
+    if (!req.body) {
+        response.json({
+            message: 'Please pass a body to the req',
+        })
+    }
+    if (!req.body.user.name) {
+        response.json({
+            message: 'Please insert a name',
+        })
+    }
+    if (!req.body.user.location || req.body.user.location.length === 0) {
+        response.json({
+            message: 'Please insert a location',
+        })
+    }
+    let id = uniqid();
+    crate.execute("INSERT INTO guestbook.posts (id, \"user\", text, created, image_ref, like_count) VALUES (?, ?, ?, CURRENT_TIMESTAMP, ?, ?)", [
+        id,
+        req.body.user,
+        req.body.text,
+        req.body.image_ref,
+        0
+    ]).then((res) => {
+        if (res.rowcount > 0) {
+            crate.execute("REFRESH TABLE guestbook.posts").then(() => {
+                crate.execute("SELECT p.*, c.name as country, c.geometry as area FROM guestbook.posts AS p, guestbook.countries AS c WHERE within(p.\"user\"['location'], c.geometry) AND p.id = ?", [id]).then((res) => {
+                    response.status(200).json(res.json);
+                });
+            });
+        } else {
+            response.json({
+                message: 'DB error',
+            })
+        }
+    });
+});
+
+// change the text of a post
+app.put('/post/:id', (req, response) => {
+    console.log(req.method + ' ' + req.url);
+
+    if (!req.body) {
+        response.json({
+            message: 'Please pass a body to the req',
+        })
+    }
+    if (!req.body.text) {
+        response.json({
+            message: 'Argument "text" is required',
+        })
+    }
+    crate.execute("UPDATE guestbook.posts SET text=? WHERE id=?", [req.body.text, req.params.id]).then((res) => {
+        if (res.rowcount > 0) {
+            crate.execute("REFRESH TABLE guestbook.posts").then(() => {
+                crate.execute("SELECT p.*, c.name as country, c.geometry as area FROM guestbook.posts AS p, guestbook.countries AS c WHERE within(p.\"user\"['location'], c.geometry) AND p.id = ?", [req.params.id]).then((res) => {
+                    response.status(200).json(res.json);
+                });
+            });
+        } else {
+            response.status(404).json({
+                message: 'Post not found'
+            })
+        }
+    });
+});
+
+// increments the number of likes for a given post
+app.put('/post/:id/like', (req, response) => {
+    console.log(req.method + ' ' + req.url);
+
+    crate.execute("SELECT * FROM guestbook.posts WHERE id=?", [req.params.id]).then((res) => {
+        if (res.rowcount > 0) {
+            crate.execute("UPDATE guestbook.posts SET like_count = like_count + 1 WHERE id=?", [req.params.id]).then((res) => {
+                if (res.rowcount > 0) {
```

Comment:
early return instead - in this example you'll see that it reduces nesting and makes code much more readable


this is the last time i'm writing it - please check for the whole file ;)